packages/plans-grid-next/src/types.ts

```diff
@@ -135,25 +134,59 @@ export interface ComparisonGridProps extends CommonGridProps {
 	selectedPlan?: string;
 }
 
-export type UseActionCallbackParams = {
+export type UseActionCallback = ( {
+	planSlug,
+	cartItemForPlan,
+	selectedStorageAddOn,
+}: {
 	planSlug: PlanSlug;
 	cartItemForPlan?: MinimalRequestCartProduct | null;
 	selectedStorageAddOn?: AddOns.AddOnMeta | null;
-};
+} ) => () => void;
 
-export type UseActionCallback = ( {
-	planSlug,
+export interface GridAction {
+	primary: {
+		text: TranslateResult;
+		callback: () => void;
+		status?: any;
```

Comment:
Need to make `status` concrete obviously and consider the default/fallback value